### PR TITLE
use omitProps everywhere with passThroughs

### LINKED ISSUE #408: Fix pass throughs on a couple components
We need to omit props on a couple components. There are probably more than two, but theses are the ones we know about right now.

appnexus_console_-_console_report_ui

### PR DESCRIPTION
**Pull request by sodiumjoe**
kind of a doozy: http://docspot.devnxs.net/projects/lucid/408-passthroughs/#/

smoke-testing help appreciated

Fixes #408

## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - ~~[ ] Firefox~~
  - ~~[ ] Safari~~
  - ~~[ ] IE11 (Win 7)~~
  - ~~[ ] Edge (Win 10)~~
- ~~[ ] Unit tests written (`common` at minimum)~~
- [x] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- ~~[ ] One core team UX approval~~

### CODE REVIEW

File: src/components/Button/Button.jsx [hunk 1]
```diff
@@ -114,7 +114,7 @@ const Button = createClass({
 
 		return (
 			<button
-				{...passThroughs}
+				{...omitProps(passThroughs, Button, ['callbackId'])}
```
Comment (sodiumjoe) [inline src/components/Button/Button.jsx]: this pattern is a little problematic, sort of couples `Button` to `ButtonGroup`. we should try to think of another way to pass arbitrary data to callbacks

File: src/components/Icon/ArrowIcon/ArrowIcon.jsx [hunk 2]
```diff
@@ -46,7 +47,8 @@ const ArrowIcon = createClass({
 
 		return (
 			<Icon
-				{...passThroughs}
+				{...omitProps(passThroughs, ArrowIcon)}
+				{..._.pick(passThroughs, _.keys(Icon.propTypes))}
```
Comment (sodiumjoe) [inline src/components/Icon/ArrowIcon/ArrowIcon.jsx]: we want to allow `props` for `Icon` to get through so `Icon` can handle them

File: src/components/Button/Button.jsx (hunk 1 above)
Comment (jondlm) [inline src/components/Button/Button.jsx]: Yeah agreed. I'm not keen on the pattern either. I recently had a little discussion on twitter about this. One solution was to use a memoized version of `partial`. I'm not sure I completely understand the implications of it, but it's food for thought.

https://twitter.com/davidlymanning/status/715993185403740160

File: src/components/RadioGroup/RadioGroup.spec.jsx [hunk 3]
```diff
@@ -48,14 +48,6 @@ describe('RadioGroup', () => {
 			});
 		});
 
-		describe('onSelect', () => {
```
Comment (sodiumjoe) [inline src/components/RadioGroup/RadioGroup.spec.jsx]: this was not a good test

1. i don't think we should test that `defaultProps` works properly
2. this was testing the exact wrong thing, i.e. that `onSelect` was getting passed to the root `span`, which we shouldn't be doing, and which `omitProps` prevents

File: src/components/Button/Button.jsx (hunk 1 above)
Comment (sodiumjoe) [inline src/components/Button/Button.jsx]: we should get @tarun in on this discussion, i think the's thought about this the most

File: src/components/ContextMenu/ContextMenu.jsx [hunk 4]
```diff
@@ -188,14 +188,16 @@ const ContextMenu = createClass({
...
 
-		const alignmentOffset = (alignment === ContextMenu.CENTER)
-			? (getAlignmentOffset(_.includes([ContextMenu.UP, ContextMenu.DOWN], direction) ? flyOutWidth : flyOutHeight))
-			: 0;
-
 		if (!flyOutPortal) {
 			return {};
 		}
 
+		const alignmentOffset = !_.isUndefined(this.props.alignmentOffset)
```
Comment (jondlm) [inline src/components/ContextMenu/ContextMenu.jsx]: I'm guessing this was the bugfix you mentioned

Comment (sodiumjoe) [inline src/components/ContextMenu/ContextMenu.jsx]: yep, wasn't setting `alignmentOffset` at all

File: src/components/DataTable/examples/2.column-groups.jsx [hunk 5]
```diff
@@ -80,7 +80,7 @@ export default React.createClass({
 		return(
 			<DataTable
 				data={data}
-				hasExtraWhitespace
```
Comment (jondlm) [inline src/components/DataTable/examples/2.column-groups.jsx]: Good catch

File: src/components/Dialog/Dialog.spec.jsx [hunk 6]
```diff
@@ -5,15 +5,24 @@ import describeWithDOM from '../../util/describe-with-dom';
 import assert from 'assert';
 
 import Dialog from './Dialog';
+import Overlay from '../Overlay/Overlay';
 
-describe('Dialog', () => {
+describe.only('Dialog', () => {
```
Comment (jondlm) [inline src/components/Dialog/Dialog.spec.jsx]: @sodiumjoe make sure to remove this :)

File: src/components/Dialog/Dialog.spec.jsx [hunk 7]
```diff
@@ -14,6 +15,14 @@ describe('Dialog', () => {
 		},
 	});
 
+	it('should pass `isModal` to underlying Overlay', () => {
```
Comment (jondlm) [inline src/components/Dialog/Dialog.spec.jsx]: Thanks for adding a test!

File: src/components/RadioGroup/RadioGroup.spec.jsx (hunk 3 above)
Comment (jondlm) [inline src/components/RadioGroup/RadioGroup.spec.jsx]: Agreed. There are probably a smattering of these in our tests.

File: src/components/Tabs/Tabs.jsx [hunk 8]
```diff
@@ -132,7 +132,7 @@ const Tabs = createClass({
 							>
 								{_.get(getFirst(tabChildProp, Tabs.Title), 'props.children', '')}
 								{isProgressive && index !== tabChildProps.length - 1 ?
-									<svg className={cx('&-Tab-arrow')} xmlns='http://www.w3.org/2000/svg' width='8px' height='28px' viewBox='0 0 8 28' >
```
Comment (jondlm) [inline src/components/Tabs/Tabs.jsx]: Man, you found a ton of goodies in this PR.

File: src/components/Button/Button.jsx (hunk 1 above)
Comment (tarun) [inline src/components/Button/Button.jsx]: here omitProps is just a call so the whole cost of creating partial function scenario most likely doesn't apply here...
as for memoization - a reselect.createSelector wrapped function may optimize - but probably not worth it at a lower level...

**Pull request merged**